packages/tss-client/src/client.ts

```diff
@@ -384,55 +373,55 @@ export class Client {
     }
 
     this._startSignTime = Date.now();
-    const sigFragmentsPromises = [];
-    for (let i = 0; i < this.precomputes.length; i++) {
-      const precompute = this.precomputes[i];
+    const sigFragments: string[] = [];
+    const fragmentPromises: Promise<string>[] = [];
+
+    for (let i = 0; i < this.parties.length; i++) {
       const party = i;
-      if (precompute === "precompute_complete") {
-        const endpoint = this.lookupEndpoint(this.session, party);
-        sigFragmentsPromises.push(
-          fetch(`${endpoint}/sign`, {
-            method: "POST",
-            headers: {
-              "Content-Type": "application/json",
-              [WEB3_SESSION_HEADER_KEY]: this.sid,
-            },
-            body: JSON.stringify({
-              session: this.session,
-              sender: this.index,
-              recipient: party,
-              msg,
-              hash_only,
-              original_message,
-              hash_algo,
-              ...additionalParams,
-            }),
+      if (party === this.index) {
+        // create signature fragment for this client
+        sigFragments.push(await tss.local_sign(msg, hash_only, this.precomputed_value));
```

Comment:
nice refactor, now code is looks cleaner and readable.

Reply:
Appreciated :)